pkg/http/http.go

```diff
@@ -262,3 +392,151 @@ func (c *Client) Discover(ctx context.Context) {
 func (c *Client) Resolve(ctx context.Context) error {
 	return c.provider.Resolve(ctx, append(c.fileSDCache.Addresses(), c.staticAddresses...))
 }
+
+// Duplicated from github.com/prometheus/common/config
+// better option would be to expose config_util.newTLSRoundTripper function from prometheus
```

Comment:
Yes, can we do that (you can contribute there, we are happy to review and merge it there)

Reply:
Great, I'll raise a PR there

Reply:
https://github.com/prometheus/common/pull/301